src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOQuerySupport.scala

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.dsde.firecloud.dataaccess
+
+import org.broadinstitute.dsde.firecloud.model._
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.elasticsearch.search.aggregations.{AggregationBuilders, Aggregations}
+import org.elasticsearch.client.transport.TransportClient
+import org.elasticsearch.action.search.{SearchRequest, SearchRequestBuilder, SearchResponse}
+import org.elasticsearch.search.aggregations.bucket.terms.Terms
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
+trait ElasticSearchDAOQuerySupport extends ElasticSearchDAOSupport {
+
+  def createListOfMusts(fields: Map[String, Seq[String]]): Seq[QueryMap] = {
+    (fields map {
+      case (k, v) => ESBool(createESShouldForTerms(k, v))
+    }).toSeq
+  }
+
+  def createESShouldForTerms(attribute: String, terms: Seq[String]): ESShould = {
+    val clauses: Seq[ESTerm] = terms map {
+      case (term: String) => ESTerm(Map(attribute -> term))
+    }
+    ESShould(clauses)
+  }
+
+  def createQuery(criteria: LibrarySearchParams): Seq[QueryMap] = {
+    (criteria.searchString, criteria.searchFields.size) match {
+      case (None | Some(""), 0) => Seq(new ESMatchAll)
+      case (None | Some(""), _) => createListOfMusts(criteria.searchFields)
+      case (Some(searchTerm: String), 0) => Seq(new ESMatch(searchTerm.toLowerCase))
+      case (Some(searchTerm: String), _) => createListOfMusts(criteria.searchFields) :+ new ESMatch(searchTerm.toLowerCase)
+    }
```

Comment:
I know I suggested this `(criteria.searchString, criteria.searchFields.size) match` approach in the first place, but as we get more and more filtering criteria, I think it'll get unworkable. Is this approach better?
```
    val search = criteria.searchString match {
      case None => Seq.empty
      case Some(searchTerm) if searchTerm.trim == "" => Seq.empty
      case Some(searchTerm) => Seq(new ESMatch(searchTerm))
    }
    
    val facetNav = criteria.searchFields match {
      case navs if navs.isEmpty => Seq.empty
      case navs => createListOfMusts(navs)
    }
    
    search ++ facetNav
```

(I removed the toLowerCase - why do we need that?)